docarray/document/mixins/video.py

```diff
@@ -110,16 +110,36 @@ def load_uri_to_video_tensor(
         import av
 
         with av.open(self.uri, **kwargs) as container:
-            if only_keyframes:
-                stream = container.streams.video[0]
-                stream.codec_context.skip_frame = 'NONKEY'
+            stream = container.streams.video[0]
+            stream.codec_context.skip_frame = 'NONKEY'
 
-            frames = []
+            key_frames = []
             for frame in container.decode(video=0):
+                img = frame.to_image()
+                key_frames.append(img)
+
+        with av.open(self.uri, **kwargs) as container:
```

Comment:
if we only want key frame we don't need to repopen the files again

Reply:
No, rather it is when `only_keyframes=true` a part of your current code is not needed it is executed but you return a result that you compute earlier so you can just skip it when you only want the keyframes